modules/util/main/index.coffee

```diff
@@ -89,6 +89,30 @@ hx.minBy = (values, f) ->
         min = v
     min
 
+hx.argmin = (values, f) ->
+  if not values? or values.length is 0 then return undefined
+
+  minIndex = 0
+  minValue = undefined
+  if f
+    minValue = f(values[0])
+    if values.length > 1
+      for i in [1...values.length] by 1
+        v = f(values[i])
+        if minValue is undefined or (v isnt undefined and v < minValue)
+          minValue = v
+          minIndex = i
+  else
+    minValue = values[0]
+    if values.length > 1
+      for i in [1...values.length] by 1
+        v = values[i]
+        if minValue is undefined or (v isnt undefined and v < minValue)
+          minValue = v
+          minIndex = i
```

Comment:
Could do, but I did it this way to avoid the extra function call when dealing with number arrays. I'll benchmark to see how much overhead that actually adds. Maybe hx.identity gets optimised away...